### PR TITLE
Change simtools package layout from 'flat' to 'src'.

### LINKED ISSUE #1261: Move to src-layout for simtools
Change simtools to src-layout as described in https://setuptools.pypa.io/en/latest/userguide/package_discovery.html#src-layout

simtools is currently organized as [flat layout](https://setuptools.pypa.io/en/latest/userguide/package_discovery.html#flat-layout), which is different to the recommendation from CTAO (see [python-template](https://gitlab.cta-observatory.org/cta-computing/documentation/python-project-template#project-structure--packaging)) and other CTA projects (e.g. [ctapipe](https://github.com/cta-observatory/ctapipe).

Being in sync with those project has the benefits of lowering the maintenance for packaging and deployment, as we can take over one-to-one the settings from other CTAO projects. 

This is triggered by the issues from python building (see  [here](https://github.com/gammasim/simtools/actions/runs/12032739454/job/33545254687#step:5:876)), by the failure of the gammasimtools feedstock (see [here](https://github.com/conda-forge/gammasimtools-feedstock/pull/1/checks?check_run_id=33554392192)), and the failure of solve this issue within a reasonable time scale.

I will make these changes, unless someone wants to become an expert in the whole packaging ecosystem.

### PR DESCRIPTION
**Pull request by GernotMaier**
Python offers two ways on how to organize a package: `flat` and `src-layout`. For background, see

- https://packaging.python.org/en/latest/discussions/src-layout-vs-flat-layout/
- https://www.pyopensci.org/python-package-guide/package-structure-code/python-package-structure.html

simtools followed until now the `flat` layout. This PR changes the layout to `src-type`.

Motivation:

- most (all?) documentation recommends `src-layout`, as it allows clearer separation between code and tests
- CTAO template recommends src layout (see  [python-template](https://gitlab.cta-observatory.org/cta-computing/documentation/python-project-template#project-structure--packaging)). Other CTA projects (e.g. [ctapipe](https://github.com/cta-observatory/ctapipe)) are using it.
- current structure means that all tests (and the test data) is added to the pypi package. This is unnecessary.

Given the issues / error messages we have in building pypi and conda packages (see  [here](https://github.com/gammasim/simtools/actions/runs/12032739454/job/33545254687#step:5:876)) and gammasimtools feedstock (see [here](https://github.com/conda-forge/gammasimtools-feedstock/pull/1/checks?check_run_id=33554392192)). None of us is an expert on packaging / building and maintenance is simplified when we are in sync with everyone else.

What changed in this PR:

- move of `simtools` directory to `src/simtools`
- adaption of directory in pyproject.toml and the `docs/source/conf.py`
- fix of 3 unit and 2 integration tests with hardwired paths
- adaption of paths in github worfklows
- replaced `Path(__file__).parent / "../simtools...` by `files("simtools") /` (otherwise schema files are not found)

Tested the python building and the warnings / error seen before (e.g.  [here](https://github.com/gammasim/simtools/actions/runs/12032739454/job/33545254687#step:5:876)) are gone. Also easy to compare when looking at:

- [code from current main with warnings](https://github.com/gammasim/simtools/actions/runs/12110724735/job/33761562283#step:3:837)
- [this branch building](https://github.com/gammasim/simtools/actions/runs/12120848556/job/33790466723#step:3:797)

As all files have been moved, this might generate some conflicts with existing PRs. But I guess there is never a good time do do this, so let's simple fix it.

Closes #1261

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -1,8 +1,8 @@
 [build-system]
 build-backend = "setuptools.build_meta"
 requires = [
-  "setuptools>=65",
-  "setuptools-scm[toml]>=6.2",
+  "setuptools>=64",
```
Comment (tobiaskleiner) [inline pyproject.toml:4]: Why lowering the version here?

Comment (GernotMaier) [reply pyproject.toml:4]: Sync with cta pyproject and ctapipe. I don't remember why we had 65 (and I don't know why 64). I trust the guys responsible for those package know more than me....

**Review by tobiaskleiner — APPROVED** [review]
@GernotMaier thanks for the implementation of the source layout. Looks all good and you can merge.

File: pyproject.toml (hunk 1 above)
Comment (tobiaskleiner) [reply pyproject.toml:4]: Ok good

**Pull request merged**